STUDIO/org.openl.rules.webstudio/src/org/openl/rules/rest/tags/TagConfigService.java

```diff
@@ -0,0 +1,132 @@
+package org.openl.rules.rest.tags;
+
+import java.util.List;
+
+import javax.ws.rs.Consumes;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.FormParam;
+import javax.ws.rs.GET;
+import javax.ws.rs.PATCH;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+import org.openl.rules.rest.SecurityChecker;
+import org.openl.rules.security.Privileges;
+import org.openl.rules.security.standalone.persistence.Tag;
+import org.openl.rules.security.standalone.persistence.TagType;
+import org.openl.rules.webstudio.service.TagService;
+import org.openl.rules.webstudio.service.TagTypeService;
+import org.openl.rules.webstudio.util.NameChecker;
+import org.openl.rules.webstudio.web.util.WebStudioUtils;
+import org.openl.util.StringUtils;
+import org.springframework.stereotype.Service;
+
+@Service
+@Path("/admin/tag-config")
+@Produces(MediaType.APPLICATION_JSON)
+@Consumes(MediaType.APPLICATION_JSON)
+public class TagConfigService {
+    private final TagTypeService tagTypeService;
+    private final TagService tagService;
+
+    public TagConfigService(TagTypeService tagTypeService, TagService tagService) {
+        this.tagTypeService = tagTypeService;
+        this.tagService = tagService;
+    }
+
+    @GET
+    @Path("/types")
+    public List<TagTypeDTO> getTypes() {
+        SecurityChecker.allow(Privileges.ADMIN);
+        return tagTypeService.getAll();
+    }
+
+    @DELETE
+    @Path("/type")
+    @Consumes(MediaType.TEXT_PLAIN)
+    public void deleteTagType(final Long id) {
+        SecurityChecker.allow(Privileges.ADMIN);
+        tagTypeService.delete(id);
+    }
+
+    @DELETE
+    @Path("/tag")
+    @Consumes(MediaType.TEXT_PLAIN)
+    public void deleteTag(final Long id) {
+        SecurityChecker.allow(Privileges.ADMIN);
+        tagService.delete(id);
+    }
+
+    @PATCH
+    @Path("/type")
+    @Consumes(MediaType.APPLICATION_FORM_URLENCODED)
+    public void updateTagType(@FormParam("id") final Long id,

```

Comment:
I think it should accept JSON request instead of form data. Form data is useless in this case

Reply:
Also ```id``` should be as path param